TileStache/Goodies/AreaServer.py

```diff
@@ -20,10 +20,18 @@
       http://localhost:8888/layer-name?width=600&height=600&xmin=-100&ymin=-100&xmax=100&ymax=100
 """
 
-from urlparse import parse_qsl
+try:
+    from urllib.parse import parse_qsl
+except ImportError:
+    # Python 2
+    from urlparse import parse_qsl
 from datetime import timedelta
 from datetime import datetime
-from StringIO import StringIO
+try:
+    from io import StringIO
```

Comment:
I think we actually need BytesIO

Reply:
Forgive my confusion, but why is StringIO then imported in the original code? Was it faulty before, or am I missing something here?

Reply:
Py2's StringIO deals with both text and byte strings, but Py3's StringIO only deals with text (unicode to be precise). We are using StringIO to write PNGs, thus we need byte support.